### PR TITLE
Flag for release blocking, document extraNames

### PR DESCRIPTION
**Pull request by zbynek**
Follow-up for https://github.com/jenkins-infra/repository-permissions-updater/pull/4108

Restricts the inconsistencies between artifact name and paths by adding a flag for release blocking and cleans the usages of wildcard paths. Documentation is updated accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,8 @@ Each file contains the following in [YAML format](https://en.wikipedia.org/wiki/
 - A `name` (typically mirrored in the file name), this is also the `artifactId` of the Maven artifact.
 - A `github` field indicating the GitHub organization and repository which is expected to produce these artifacts.
 - A set of paths, usually just one. These correspond to the full Maven coordinates (`groupId` and `artifactId`) used for the artifact. Since Jenkins plugins can change group IDs and are still considered the same artifact, multiple entries are possible.
+- Optional `releaseBlocked` flag. When set to true, 
+ artifact(s) described in this file cannot be pushed to Artifactory.
```
Comment (daniel-beck) [inline README.md:40]:
```suggestion
- Optional `releaseBlocked` flag. When set to true, artifact(s) described in this file cannot be pushed to Artifactory. This is only used in rare circumstances, e.g., by the Jenkins security team if plugin code contains unreleased security issues.
```

File: README.md [hunk 2]
```diff
@@ -86,6 +88,14 @@ Changing the `paths` or modifying the `<artifactId>` in the plugin `pom.xml` is
...
 Altering the `artifactId` means changing the identifier by which the Jenkins plugin manager differentiates one plugin from others, and will cause chaos for users who have already installed it under the old name.
 
+### Multiple plugins in single configuration file
+
+If multiple closely related plugins are located in the same GitHub repository,
+have the same maintainer and the same issue tracker, they can use a single configuration file.
+To define multiple plugins in a single file, please
+- add an entry to `paths` that includes a `*` character, this path has to match the paths of all included plugins, but should not be too generic.
+- add a new property `extraNames`, it should contain the list of names of all the plugins managed by this configuration file other than the main plugin defined by `name`.
+
```
Comment (daniel-beck) [inline README.md:98]: This document (mostly) uses sentence-per-line formatting.
```suggestion
If multiple closely related plugins are located in the same GitHub repository, have the same maintainer and the same issue tracker, they can use a single configuration file.
To define multiple plugins in a single file, please do the following:
- Add an entry to `paths` that includes a `*` character, this path has to match the paths of all included plugins, but should not be too generic.
- Add a new property `extraNames`, it should contain the list of names of all the plugins managed by this configuration file other than the main plugin defined by `name`.

```

File: src/test/java/io/jenkins/infra/repository_permissions_updater/ArtifactoryPermissionUpdaterTest.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package io.jenkins.infra.repository_permissions_updater;
+
+public class ArtifactoryPermissionUpdaterTest {
```
Comment (timja) [inline src/test/java/io/jenkins/infra/repository_permissions_updater/ArtifactoryPermissionUpdaterTest.java:3]: did you mean to commit this?

File: README.md [hunk 4]
```diff
@@ -200,6 +209,8 @@ The list is sorted in descending order of preference.
 The first issue tracker in the list with a `reportUrl` should be presented as the primary (or only) option for reporting issues.
 Further issue trackers are mostly provided as a reference, e.g. when listing existing issues, although a different issue tracker with `reportUrl` can be linked if users provide a preference for a specific kind of issue tracker.
 
+## Multiple
```
Comment (timja) [inline README.md:212]: did you mean to have this heading:

Comment (zbynek) [reply README.md]: thanks, deleted

**Review by timja — APPROVED** [review]

**Pull request merged**